mips/copy.S

```diff
@@ -0,0 +1,155 @@
+/* Mostly copied from FreeBSD mips/support.S */
+
+#include <errno.h>
+#include <mips/asm.h>
+#include <mips/regdef.h>
+#include <mips/pcpu.h>
+
+#include "assym.h"
+
+	.set	noreorder		# Noreorder is default style!
+
+/*
+ * int copystr(const void *kfaddr, void *kdaddr, size_t len, size_t *lencopied)
+ *
+ * Copy a NIL-terminated string, at most maxlen characters long.  Return the
+ * number of characters copied (including the NIL) in *lencopied.  If the
+ * string is too long, return -ENAMETOOLONG; else return 0.
```

Comment:
I'm not sure I like this error code, `ENAMETOOLONG` refers to filenames. Maybe `ENOMEM` ("Not enough space") would be better?